pkg/controller/backup_storage_location_controller.go

```diff
@@ -0,0 +1,210 @@
+/*
+Copyright 2020 the Velero contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"encoding/json"
+	"strings"
+	"time"
+
+	jsonpatch "github.com/evanphx/json-patch"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	velerov1api "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	velerov1client "github.com/vmware-tanzu/velero/pkg/generated/clientset/versioned/typed/velero/v1"
+	velerov1listers "github.com/vmware-tanzu/velero/pkg/generated/listers/velero/v1"
+	"github.com/vmware-tanzu/velero/pkg/persistence"
+	"github.com/vmware-tanzu/velero/pkg/plugin/clientmgmt"
+)
+
+type backupStorageLocationController struct {
+	*genericController
+
+	namespace                    string
+	defaultBackupLocation        string
+	defaultStoreValidationPeriod time.Duration
+	backupLocationClient         velerov1client.BackupStorageLocationsGetter
+	backupStorageLocationLister  velerov1listers.BackupStorageLocationLister
+	newPluginManager             func(logrus.FieldLogger) clientmgmt.Manager
+	newBackupStore               func(*velerov1api.BackupStorageLocation, persistence.ObjectStoreGetter, logrus.FieldLogger) (persistence.BackupStore, error)
+}
+
+func NewBackupStorageLocationController(
+	namespace string,
+	defaultBackupLocation string,
+	defaultStoreValidationPeriod time.Duration,
+	backupLocationClient velerov1client.BackupStorageLocationsGetter,
+	backupStorageLocationLister velerov1listers.BackupStorageLocationLister,
+	newPluginManager func(logrus.FieldLogger) clientmgmt.Manager,
+	logger logrus.FieldLogger,
+) Interface {
+	if defaultStoreValidationPeriod <= 0 {
```

Comment:
if `defaultStoreValidationPeriod == 0` doesn't that mean validation is disabled?

Reply:
Yes so, this is confusing because of two flags with the same name. Mainly I set this up following the backup-sync-controller. We can remove this or change the condition here. 

This right here is a flag on the server. It is set so that `IF` `location.Spec.StoreValidationPeriod` further below is not set, we'll use this value. And so, if there wasn't a value passed thru the server either, it is set to `1min`.

We could change this to match the BSL flag, in other words, if `<=0` then don't ever validate. But I don't think we want to never validate. And I know that's not what you are saying either. Since we do have to validate at some interval, we might as well make it configurable thru the server.

Reply:
Addendum: *we don't want to never validate a BSL >unless< the user specifically configures it that way.